pkg/controller/datavolume-conditions.go

```diff
@@ -0,0 +1,159 @@
+					conditions = updateReadyCondition(conditions, corev1.ConditionFalse, "", transferRunning)
+				} else if strings.ToLower(sourceRunningVal) == "false" {
+					// target running, source not running, overall not running.
+					conditions = updateCondition(conditions, cdiv1.DataVolumeRunning, corev1.ConditionFalse, anno[AnnSourceRunningConditionMessage], anno[AnnSourceRunningConditionReason])
+				} else {
+					conditions = updateCondition(conditions, cdiv1.DataVolumeRunning, corev1.ConditionUnknown, anno[AnnSourceRunningConditionMessage], anno[AnnSourceRunningConditionReason])
+				}
+			} else {
+				conditions = updateCondition(conditions, cdiv1.DataVolumeRunning, corev1.ConditionTrue, anno[AnnRunningConditionMessage], anno[AnnRunningConditionReason])
+				conditions = updateReadyCondition(conditions, corev1.ConditionFalse, "", transferRunning)
+			}
+		} else if strings.ToLower(val) == "false" {
+			if sourceRunningVal, ok := anno[AnnSourceRunningCondition]; ok {
+				if strings.ToLower(sourceRunningVal) == "true" {
+					conditions = updateCondition(conditions, cdiv1.DataVolumeRunning, corev1.ConditionFalse, anno[AnnRunningConditionMessage], anno[AnnRunningConditionReason])
+				} else if strings.ToLower(sourceRunningVal) == "false" {
+					// target not running and source not running, overall not running.
+					conditions = updateCondition(conditions, cdiv1.DataVolumeRunning, corev1.ConditionFalse, fmt.Sprintf("%s and %s", anno[AnnRunningConditionMessage], anno[AnnSourceRunningConditionMessage]), fmt.Sprintf("%s and %s", anno[AnnRunningConditionReason], anno[AnnSourceRunningConditionReason]))
+				} else {
+					conditions = updateCondition(conditions, cdiv1.DataVolumeRunning, corev1.ConditionUnknown, fmt.Sprintf("%s and %s", anno[AnnRunningConditionMessage], anno[AnnSourceRunningConditionMessage]), fmt.Sprintf("%s and %s", anno[AnnRunningConditionReason], anno[AnnSourceRunningConditionReason]))
+				}
+			} else {
+				conditions = updateCondition(conditions, cdiv1.DataVolumeRunning, corev1.ConditionFalse, anno[AnnRunningConditionMessage], anno[AnnRunningConditionReason])
+			}
+		} else {
+			conditions = updateCondition(conditions, cdiv1.DataVolumeRunning, corev1.ConditionUnknown, anno[AnnRunningConditionMessage], anno[AnnRunningConditionReason])
+		}
+	} else {
+		conditions = updateCondition(conditions, cdiv1.DataVolumeRunning, corev1.ConditionUnknown, anno[AnnRunningConditionMessage], anno[AnnRunningConditionReason])
+	}
+	return conditions
+}
+
+func updateReadyCondition(conditions []cdiv1.DataVolumeCondition, status corev1.ConditionStatus, message, reason string) []cdiv1.DataVolumeCondition {
+	return updateCondition(conditions, cdiv1.DataVolumeReady, status, message, reason)
+}
+
+func updateBoundCondition(conditions []cdiv1.DataVolumeCondition, pvc *corev1.PersistentVolumeClaim) []cdiv1.DataVolumeCondition {
```

Comment:
Well the annotation can also be the bound status of the source PVC, if we are cloning. But it might be a good idea to put the logic in the respective controller instead of the data volume controller, and have the data volume controller just read the annotation.